sys/include/atomic_utils.h

```diff
@@ -0,0 +1,1053 @@
+/*
+ * Copyright (C) 2020 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    sys_atomic_utils    Utility functions for atomic access
+ * @ingroup     sys
+ *
+ * This modules adds some utility functions to perform atomic accesses.
+ *
+ * # Usage
+ *
+ * The atomic utilitys allow atomic access to regular variables.
+ *
+ *  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~{.c}
+ *  uint32_t global_counter;
+ *
+ *  void irq_handler(void)
+ *  {
+ *      // No need to use atomic access in IRQ handlers: RIOT does not
+ *      // enable nested interrupts, so IRQ handlers will not be interrupted
```

Comment:
(this is not entirely true anymore, eg., PendSV is now lower priority than the other ISRs)

Reply:
Hmm, as a result every memory that is also accessed by PendSV must then use atomic accesses as well.

Let me rephrase this to be extra careful. If other ISRs do not touch the same variable, no issue would occur with nesting IRQs. And likely this is the default case.